logstash-core-event-java/src/main/java/org/logstash/Accessors.java

```diff
@@ -34,10 +34,15 @@ public Object del(String reference) {
                 return ((Map<String, Object>) target).remove(field.getKey());
             } else if (target instanceof List) {
                 int i = Integer.parseInt(field.getKey());
-                if (i < 0 || i >= ((List) target).size()) {
-                    return null;
+                int size = ((List) target).size();
+                if (i >= size || i < -size) {
```

Comment:
Yeah, I was thinking of refactoring this to add a method that turns `i` into a list offset anyway, that way there's less branches for these bounds checks all over the place.

I can do that.